Orange/canvas/scheme/widgetsscheme.py

```diff
@@ -627,6 +635,15 @@ def widget_processing_state(self, widget):
         """
         return self.__widget_processing_state[widget]
 
+    def show_widgets_on_top_changed(self):
+        """
+        `Float Widgets on Top` menu option has changed.
+
+        Update the flag on existing widgets.
+        """
```

Comment:
Change this to a setter (`def set_show_widgets_on_top(self, state):`) record the  state in an instance attribute, and use that instead of querying QSettings in  `__set_float_on_top`, so the state is local to the workflow and not shared between them.